### DISCUSSION TITLE
DOUBT: How to recover routing table after deep sleep

### DISCUSSION THREAD

**Discussion by vinieich** [LoRaMesher/LoRaMesher#36] [General]
Hello
This is a very interesting library, nice work.
I am engineer student and I'm trying to use this library with some heltec V2 LoRa.
I'm only doing the firsts tests now but there is something that I would like to ask if it's not any problem.

Is there any way to recover from esp32 deep sleep and send a packet without the routing table? Like if there is a node that is assured to be on and in range. Or maybe is there a way to export the routing table? So then it can be saved in the RTC memory that is saved on deep sleep wake up.

Thanks in advance and sorry for any grammar mistakes, english is not my primary language.

**Comment by Jaimi5** [comment]
Hi vinieich!

It's being a while that I am thinking about storing the Routing Table! I don't really know if all the routing table could be stored in the RTC or SPIFFS memory. Have you more experience in this field?
If it can be stored, I can do a came around that when putting the library in sleep mode, it could write the memory in some of these memories. 

Moreover, with the heltec V2 LoRa the library works correctly? I don't have this device and cannot test it!

PD: I am not native in English too :P

**Reply by marceloroberti** [reply]
Hey @vinieich and @Jaimi5 — hope you’re both doing well!

Have you found a good way to save the routing table? I’m shooting for maximum power-saving: build the table once in a “config” mode, store it, then let each node drop into deep sleep whenever it’s not transmitting. If I ever move or add nodes, I’d just go back into config, rebuild the table, and save again.

Have you tried something similar or have any hints?

Thanks, and congratulations for the work on the library!

**Reply by Jaimi5** [reply]
I haven’t tried that approach myself, but saving the routing table should be fairly straightforward to implement. That said, I don’t plan to add it in the current version of LoRaMesher. However, I’m actively working on a new version where I could definitely consider adding this functionality.

If you're interested in implementing it yourself, a good place to start would be the `services/RoutingTableService.h` file. I’d be happy to help guide you through the process if you decide to give it a shot.

Also, feel free to join our Discord server—it's a much easier place to discuss ideas and implementation details in real time. And thanks again for using the library!

**Reply by marceloroberti** [reply]
Thanks for the hint! I’ve used NVS on the ESP32 before to save settings between reboots, so I’ll give it a shot here. I’ll post back once I see if the same approach works with the table and deep sleep. If so, I would be happy to share with you.
